pkg/controller/backup-operator/sync.go

```diff
@@ -119,15 +121,21 @@ func (b *Backup) handleBackup(spec *api.BackupSpec) (*api.BackupStatus, error) {
 		return nil, err
 	}
 
+	// When BackupPolicy.Timeout <= 0, use default DefaultBackupTimeout.
+	backupTimeout := time.Duration(constants.DefaultBackupTimeout)
+	if spec.BackupPolicy != nil && spec.BackupPolicy.Timeout > 0 {
+		backupTimeout = time.Duration(spec.BackupPolicy.Timeout) * time.Second
+	}
+
 	switch spec.StorageType {
 	case api.BackupStorageTypeS3:
-		bs, err := handleS3(b.kubecli, spec.S3, spec.EtcdEndpoints, spec.ClientTLSSecret, b.namespace)
+		bs, err := handleS3(b.kubecli, spec.S3, backupTimeout, spec.EtcdEndpoints, spec.ClientTLSSecret, b.namespace)
```

Comment:
I rethink about it.
Instead of defining the timeout to be "saving backup", we should define the timeout to be the entire process. Each type could spend different amount of time doing other work not on saving backup. As a user, I might just define 60s and want the backup to be made around that limit without caring about getting secret, etc.

Reply:
In code:

```
ctx, cancel := context.WithTimeout(context.Background(), backupTimeout)
handleS3(ctx)
```

Reply:
@hongchaodeng yeah, that's my original thinking. However, the issue with an timeout for entire process is that kube client doesn't take-in context so I can't include kubecli activities such as retrieving secret as part of the total timeout. Hence,  the `Timeout` only applies to retrieving+saving backup.